# Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

In [46]:
import numpy as np

1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.Найти доверительный интервал для оценки математического ожидания с надежностью 0.95,если выборочная средняя M = 80, а объем выборки n = 256.

В данном случае известно среднеквадратическое отклонение генеральной совокупности, значит необходимо воспользоваться таблицей стандартного нормального распределения. Из этой таблицы можно получить информацию, что левее значения $z=1.96$ находится $\frac{\alpha}{2} = \frac{1-0.95}{2} = 2.5\%$ совокупности. Тогда доверительный интервал будет выглядеть следующим образом:
### $$\mu \in [80-1.96\cdot\frac{16}{\sqrt{256}};80+1.96\cdot\frac{16}{\sqrt{256}}] \Rightarrow \mu \in [78.04;81.96]$$

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,получены опытные данные:6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1. Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,оценить истинное значение величины X при помощи доверительного интервала, покрывающего этозначение с доверительной вероятностью 0,95.

In [47]:
X_values = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

В данном задании нам неизвестно среднеквадратическое отклонение генеральной совокупности, значит воспользуемся распределение Стьюдента для определения доверительного интервала. Также потребуется несмещенная оценка среднеквадратического отклонения.


$$t_{табл} = 2.262$$

In [48]:
sigma = np.std(X_values, ddof=1)
sigma

0.4508017549014448

In [49]:
X_mean = np.mean(X_values)
X_mean

6.590000000000001

Тогда,
### $$X \in [6.59 - 2.262\cdot\frac{0.4508}{\sqrt{10}};6.59 + 2.262\cdot\frac{0.4508}{\sqrt{10}}] \Rightarrow X \in [6.2675; 6.9125]$$

In [50]:
print(X_mean-sigma*2.262/np.sqrt(10))
print(X_mean+sigma*2.262/np.sqrt(10))

6.267538255912426
6.912461744087575


3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметроказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

Дисперсия генеральной совокупности известна, значит воспользуемся таблицей стандартного нормального распределения. Имеем следующие гипотезы:
### $$H_0: \mu=\mu_0$$
### $$H_1: \mu>\mu_0$$
Воспользуемся Z-тестом, рассчитаем значение Z:
### $$Z_р = \frac{\overline{X}-\mu_0}{\frac{\sigma}{\sqrt n}} = \frac{17.5-17}{0.2} = 2.5$$
В то время как табличное $Z_{табл}$ для данного уровня $\alpha$ равно 1.645. $Z_p > Z_{табл}\Rightarrow$ нулевая гипотеза о равенстве диаметра шариков 17мм. может быть отвергнута.

4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [51]:
X_values = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
X_mean = np.mean(X_values)
sigma = np.std(X_values, ddof=1)
print(f'Среднее значение: {X_mean}.')
print(f'Среднеквадратическое отклонение: {sigma}.')

Среднее значение: 198.5.
Среднеквадратическое отклонение: 4.453463071962462.


В данном случае нам неизвестна дисперсия генеральной совокупности, значит необходимо воспользоваться распределением Стьюдента. Гипотезы выглядят следующим образом:
### $$H_0: \mu=\mu_0$$
### $$H_1: \mu\neq\mu_0$$
Тогда расчетное значение t-статистики будет равно:
### $$t_р=\frac{\overline{X}-\mu_0}{\frac{\sigma}{\sqrt n}} \approx -1.0651$$
Табличное значение для 9 степеней свободы будет равно $t_{табл} = 3.25$. Так как $|t_р| < |t_{табл}|$, то нулевая гипотеза не может быть отвергнута.

In [52]:
print((X_mean-200)/(sigma/np.sqrt(10)))

-1.0651074037450896
